marklogic-unit-test-modules/src/main/ml-modules/root/test/test-helper.xqy

```diff
@@ -665,30 +663,30 @@ declare function helper:log($items as item()*)
 
 declare function helper:list-from-database(
   $database as xs:unsignedLong,
-  $root as xs:string,
-  $suite as xs:string?,
-  $file-type as xs:string)
+  $path as xs:string)
 as xs:string*
 {
-  if ($database = 0) then
-    helper:list-from-filesystem(fn:string-join(($root, "test", $file-type, $suite), $helper:DIRECTORY-SEPARATOR))
-  else
-    xdmp:eval(
-      'xquery version "1.0-ml";
-      declare variable $PATH as xs:string external;
-      try { cts:uris((), (), cts:directory-query($PATH, "infinity")) }
-      catch ($ex) {
-        if ($ex/error:code ne "XDMP-URILXCNNOTFOUND") then xdmp:rethrow()
-        else xdmp:directory($PATH, "infinity")/xdmp:node-uri(.) }',
-      (xs:QName('PATH'),
-      fn:concat($root, 'test/', $file-type, '/', ($suite, '')[1])),
-      <options xmlns="xdmp:eval"><database>{$database}</database></options>)
+  (: Add trailing '/' if missing :)
+  let $path := fn:replace($path, "([^/])$", "$1/")
+  return
+    if ($database = 0) then
+      let $directory-separator := if (xdmp:platform() eq "winnt") then "\\" else "/"
+      return helper:list-from-filesystem(fn:replace($path, "(/|\\)", $directory-separator))
+    else
+      xdmp:eval(
```

Comment:
We should consider replacing `xdmp:eval()` with `xdmp:invoke-function()`. Maybe not in this set of commits, to minimize changes during the refactor, but something to consider for future work.  https://www.marklogic.com/blog/first-class-functions/